compiler/compiler.go

```diff
@@ -99,13 +104,22 @@ func CompileStringWithCache(s string) (string, error) {
 		return "", err
 	}
 
-	model := fileSystem.AllFiles()[0]
+	model := fileSystem.Model("____")
```

Comment:
What does this do, ooi?

Reply:
It's because this function creates a virtual file in the file system object with the name `____`. Previously it was the only file in the filesystem, so it could just index into it.

Now we're also adding the precompiled macro file as well, so instead I'm pulling the file out by filename;

https://github.com/monzo/ddbt/blob/2b45a2f5b85924afbed0e3ddce4b6140bea81c93/compiler/compiler.go#L101-L111